tools/autograd/templates/Functions.cpp

```diff
@@ -88,8 +88,8 @@ int64_t _safe_size(IntList sizes, IntList dim) {
   return size;
 }
 
-Tensor norm_backward(const Tensor & grad, const Tensor & self, const Scalar & p_, const Tensor & norm) {
-  double p = p_.toDouble();
+Tensor norm_backward(const Tensor & grad, const Tensor & self, const optional<Scalar> & p_, const Tensor & norm) {
+  double p = p_->toDouble();
```

Comment:
This is a backward function, since p will also be specified in the forward call (either a value passed by the caller or the default 2), the backward function will always get the p_ I suppose. To make sure we get it, I will add a `value_or` here to specific the default value as well.

Reply:
the question I had was whether the above logic was right.  I don't think it is because the user can _explicitly_ pass a nullopt in C++.

Reply:
Yes that was sketchy actually, I updated it with the `value_or` semantic so that the case of p is not specified is correctly handled. Please review again when you have time :)